File: v1/code/go/src/jvmgo/ch06/instructions/extended/goto_w.go
package extended

import "jvmgo/ch06/instructions/base"
import "jvmgo/ch06/rtda"

// Branch always (wide index)
type GOTO_W struct {
	offset int
}

func (self *GOTO_W) FetchOperands(reader *base.BytecodeReader) {
	self.offset = int(reader.ReadInt32())
}
func (self *GOTO_W) Execute(frame *rtda.Frame) {
	base.Branch(frame, self.offset)
}

File: v1/code/java/jvmgo_java/ch05/src/main/java/com/github/jvmgo/instructions/extended/gotow.java
package com.github.jvmgo.instructions.extended;

import com.github.jvmgo.instructions.base.BranchInstruction;
import com.github.jvmgo.instructions.base.BytecodeReader;
import com.github.jvmgo.rtda.Frame;

public class gotow extends BranchInstruction {

    @Override
    public int getOpCode() {
        return 0xc8;
    }

    @Override
    public void fetchOperands(BytecodeReader reader) throws Exception {
        offset = reader.read32();
    }

    @Override
    public void execute(Frame frame) throws Exception {
        branch(frame, offset);
    }
}
